src/middlewared/middlewared/plugins/tunables.py

```diff
@@ -25,212 +26,110 @@ class TunableService(CRUDService):
+        Str('var', required=True),
         Str('value', required=True),
-        Str('type', enum=TUNABLE_TYPES, required=True),
+        Str('type', enum=TUNABLE_TYPES, default='SYSCTL', required=True),
         Str('comment'),
         Bool('enabled', default=True),
         register=True
     ))
     async def do_create(self, data):
         """
         Create a Tunable.
+        """
+        verrors = ValidationErrors()
+        if await self.middleware.call('tunable.query', [('var', '=', data['var'])]):
+            verrors.add('tunable.create', f'Tunable {data["var"]!r} already exists in database.', errno.EEXIST)
 
-        `var` represents name of the sysctl/loader/rc variable.
+        if data['var'] not in await self.middleware.call('tunable.get_system_tunables'):
+            verrors.add('tunable.create', f'Tunable {data["var"]!r} does not exist in kernel.', errno.ENOENT)
 
-        `type` for SCALE should be one of the following:
-        1) SYSCTL     -     Configure `var` for sysctl(8)
-        """
-        await self.clean(data, 'tunable_create')
-        await self.validate(data, 'tunable_create')
-        await self.lower(data)
-
-        data['id'] = await self.middleware.call(
-            'datastore.insert',
-            self._config.datastore,
-            data,
-            {'prefix': self._config.datastore_prefix}
-        )
+        if not (orig_value := (await self.middleware.call('tunable.get_original_value', data['var']))):
+            verrors.add('tunable.create', f'Unable to determine original value of {data["var"]!r}')
 
-        await self.middleware.call('service.reload', data['type'])
+        verrors.check()
 
-        return await self.get_instance(data['id'])
+        data['orig_value'] = orig_value
 
-    async def do_update(self, id, data):
+        if (comment := data.get('comment', '').strip()):
+            data['comment'] = comment
+
+        _id = await self.middleware.call(
+            'datastore.insert', self._config.datastore, data, {'prefix': self._config.datastore_prefix}
+        )
+        await self.middleware.call('service.restart', 'sysctl')
+        return await self.get_instance(_id)
+
+    async def do_update(self, _id, data):
         """
         Update Tunable of `id`.
         """
-        old = await self.get_instance(id)
+        old = await self.get_instance(_id)
 
         new = old.copy()
         new.update(data)
+        if old == new:
+            # nothing updated so return early
+            return old
 
-        await self.clean(new, 'tunable_update', old=old)
-        await self.validate(new, 'tunable_update')
+        if (comment := data.get('comment', '').strip()) and comment != new['comment']:
+            new['comment'] = comment
 
-        await self.lower(new)
+        if not new['enabled'] and new['orig_value']:
+            # before we delete from db, let's set the tunable back to it's original value
```

Comment:
What are we deleting from the database ?

Reply:
copy and paste error 😄